aiohttp/web_urldispatcher.py

```diff
@@ -639,59 +639,67 @@ def __iter__(self) -> Iterator[AbstractRoute]:
 
     async def _handle(self, request: Request) -> StreamResponse:
         rel_url = request.match_info["filename"]
+        filename = Path(rel_url)
+        if filename.anchor:
+            # rel_url is an absolute name like
+            # /static/\\machine_name\c$ or /static/D:\path
+            # where the static dir is totally different
+            raise HTTPForbidden()
+
+        unresolved_path = self._directory.joinpath(filename)
+        loop = asyncio.get_running_loop()
+        return await loop.run_in_executor(
+            None, self._resolve_path_to_response, unresolved_path
+        )
+
+    def _resolve_path_to_response(self, unresolved_path: Path) -> StreamResponse:
+        """Takes the unresolved path and queries the file system to form a response."""
+        # Check for access outside the root directory. For follow symlinks, URI
+        # cannot traverse out, but symlinks can. Otherwise, no access outside
+        # root is permitted.
         try:
-            filename = Path(rel_url)
-            if filename.anchor:
-                # rel_url is an absolute name like
-                # /static/\\machine_name\c$ or /static/D:\path
-                # where the static dir is totally different
-                raise HTTPForbidden()
-            unresolved_path = self._directory.joinpath(filename)
             if self._follow_symlinks:
                 normalized_path = Path(os.path.normpath(unresolved_path))
                 normalized_path.relative_to(self._directory)
-                filepath = normalized_path.resolve()
+                file_path = normalized_path.resolve()
             else:
-                filepath = unresolved_path.resolve()
-                filepath.relative_to(self._directory)
-        except (ValueError, FileNotFoundError) as error:
-            # relatively safe
-            raise HTTPNotFound() from error
-        except HTTPForbidden:
-            raise
-        except Exception as error:
-            # perm error or other kind!
-            request.app.logger.exception(error)
+                file_path = unresolved_path.resolve()
+                file_path.relative_to(self._directory)
+        # TODO(PY39): Remove OSError to appease Windows
```

Comment:
It's when 3.9 is the lowest supported (which seems to be the syntax used elsewhere in the code?).  Seems silly to me to add a conditional for the Windows 3.8 bug just for a few months?

Reply:
I updated the comment for clarity.

Reply:
We don't really use FIXME comments much. If somebody adds them, they just remain in the code base to rot. So I prefer actual code. Plus, this way, the CI would be testing both behavior branches, which is more future-proof. I'd rather keep a constant next to https://github.com/aio-libs/aiohttp/blob/3fbb093/aiohttp/helpers.py#L67 so that it'd be easier to track places that need to be cleaned up when the time comes.

Reply:
Done.  I also misspoke and it affects 3.8 and 3.9 on Windows.

Reply:
thanks!

Reply:
It's good that you moved it into a conditional as code — I didn't realize it was one exception or the other. It seemed like one was just conditionally added to the rest.

Reply:
Windows throws the same error across them all: `WinError 1921`.  The difference is that 3.10+ `pathlib` catches it and correctly re-raises as `RuntimeError`.

Reply:
> We don't really use FIXME comments much. If somebody adds them, they just remain in the code base to rot.

This convention is probably from me. I check and follow all the steps each time we drop a Python release. So, as long as it contains `PY39` or whatever, it will get cleaned up.